### ISSUE TITLE
front-page gallery brains are all white

### ISSUE THREAD

**Issue by drammock** [mne-tools/mne-python#10939]
will perform a `git bisect` today to find the culprit.

**Comment by larsoner** [comment]
Could be mesa / pyvista / pyvistaqt versions

You can browse the mne-tools.github.io repo history to find the last commit that produced a good image then compare the sys_info of that build to the next

**Comment by drammock** [comment]
Here it is:

https://github.com/mne-tools/mne-tools.github.io/commit/33b8ba9d93a8c821b8143ce87573a05976047f62#diff-4a8a84a3b9f54419858f59df784f04ad912d4f9d4b2ee04a227adcd01d8474c7

it's not just the front-page images either, it's basically all the brains.

Maybe I'm doing it wrong, but I can't see much that is different in the environment (I'm comparing the last 2 `build_docs_main` jobs where things weren't skipped): [this one](https://app.circleci.com/pipelines/github/mne-tools/mne-python/15253/workflows/93f86ae2-00cd-48e8-b8a7-2a86df7355b0/jobs/46770) versus [this one](https://app.circleci.com/pipelines/github/mne-tools/mne-python/15262/workflows/f2739a94-64a9-4f99-b131-348f659d5e6f/jobs/46788) and looking at the `list packages` step.  (IIRC, cloning the website repo takes forever, which is why I looked at the Circle UI first).

Here is what differs between those two builds:

```console
$ diff 46770.txt 46788.txt 
28c28
< darkdetect                    0.7.0
---
> darkdetect                    0.7.1
38c38
< fastjsonschema                2.15.3
---
> fastjsonschema                2.16.1
47c47
< imageio                       2.19.3
---
> imageio                       2.19.5
```

To me this suggests that is either `pyvista` or `pyvistaqt` (as you already said), since those are both using `dev` versions, something could have changed and it wouldn't be apparent from the way I'm looking.

**Comment by larsoner** [comment]
Could be. Or it's like you had in mind originally, some change in MNE-Python between those two builds. They are nightly, so we'd have to see what MNE-Python `main` commits happened between when those two builds began. The build time should come up very close to when they begin, so that would be pretty close to in between between `2022-07-16T06:19:51.130106+00:00` and `2022-07-19T06:19:00.912768+00:00`

**Comment by drammock** [comment]
yeah, but... there's very little that happened in that time frame on our side.
```
c6a86f15b (HEAD -> main, upstream/main) (5 hours ago) noreply@github.com Improve documentation of picks for IC (#10938)
e8ded9719 (24 hours ago) noreply@github.com [DOC] Fix ICA plot_sources "picks" documentation (#10936)
6955f977a (2 days ago) noreply@github.com MAINT: Fix CIs (#10933)
b733f72c4 (6 days ago) noreply@github.com fix eeglab export rawarray filenames (#10927)
```

worse, I'm struggling to get editable installs of pyvista and pyvistaqt, because of the error that @alexrockhill was hitting recently on the CIs:
`ERROR: Cannot uninstall 'vtk'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.`

Will keep hacking at it

**Comment by alexrockhill** [comment]
> yeah, but... there's very little that happened in that time frame on our side.
> 
> ```
> c6a86f15b (HEAD -> main, upstream/main) (5 hours ago) noreply@github.com Improve documentation of picks for IC (#10938)
> e8ded9719 (24 hours ago) noreply@github.com [DOC] Fix ICA plot_sources "picks" documentation (#10936)
> 6955f977a (2 days ago) noreply@github.com MAINT: Fix CIs (#10933)
> b733f72c4 (6 days ago) noreply@github.com fix eeglab export rawarray filenames (#10927)
> ```
> 
> worse, I'm struggling to get editable installs of pyvista and pyvistaqt, because of the error that @alexrockhill was hitting recently on the CIs: `ERROR: Cannot uninstall 'vtk'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.`
> 
> Will keep hacking at it

I'm took a break from trying to fix that, maybe will just wait for the patch release so if you figure it out that would be fantastic

**Comment by drammock** [comment]
update: I got around the `cannot uninstall vtk` problem, by creating a fresh environment with `vtk` removed from from `environment.yml`.  Then when I install pyvista, it pulls vtk 9.2.0rc1. I tried bisects in our repo, and in dev installs of pyvista and pyvistaqt and it's all `bad` even if I go earlier than the last `good` dev docs build date, so I'm betting it's the vtk release candidate (which pyvista is explicitly asking for) that is the culprit.

https://github.com/pyvista/pyvista/blob/2c52ffda4a0cf099c2c8950677492542698e4642/setup.py#L22

**Comment by drammock** [comment]
force-downgrading to vtk 9.1.0 did not fix it :(

**Comment by larsoner** [comment]
To me it's a bug that pyvista dev requires an RC. It shouldn't even require 9.2 when it comes out, that's too short a support window

I can bisect pyvista and/or VTK next week to debug

**Comment by drammock** [comment]
ok, downgrading to pyvista 0.35.1 solves it. Here is the diff between 0.35.1 and 0.35.2:

https://github.com/pyvista/pyvista/compare/v0.35.1...v0.35.2

I'm going to step away from this for now, after (temporarily) pinning pyvista. Hopefully @larsoner or I will have time next week to dig into the root cause.

**Comment by larsoner** [comment]
Argh @hoechenberger we need this for 1.1 as well. I'll try to get to it today

**Comment by larsoner** [comment]
The bug was introduced by https://github.com/pyvista/pyvista/pull/2980, now I need to figure out how to work around it here or in PyVista...

**Comment by larsoner** [comment]
Okay I think this our last blocker. The PyVista folks said that they plan to patch and release quickly, which for them usually does mean days (maybe today). So hopefully they do that, then we wait for the patch release to hit CDNs, then we can build installers and release!

**Comment by larsoner** [comment]
https://github.com/conda-forge/pyvista-feedstock/pull/67 might do it but https://github.com/pyvista/pyvista/issues/3116 might stand in the way

**Comment by akaszynski** [comment]
This will be addressed by https://github.com/pyvista/pyvista/pull/3117.

**Comment by larsoner** [comment]
This has been fixed by a new PyVista release of 0.36.1 !
